File: leetcode-submissions/1605_Find_Valid_Matrix_Given_Row_and_Column_Sums.cpp
class Solution {
public:
    vector<vector<int>> restoreMatrix(vector<int>& rowSum, vector<int>& colSum) {
        vector<vector<int>>matrix(rowSum.size(),vector<int>(colSum.size(),0));
        int i=0,j=0;
        int n=rowSum.size(),m=colSum.size();
        while(i<n and j<m){
            int minele=min(rowSum[i],colSum[j]);
            rowSum[i]-=minele;
            colSum[j]-=minele;
            matrix[i][j]=minele;
            if(rowSum[i]==0){
                i++;
            }else if(colSum[j]==0){
                j++;
            }else{
                i++;
                j++;
            }

        }
        return matrix;
    }
};

File: leetcode-submissions/1605_Find_Valid_Matrix_Given_Row_and_Column_Sums.ts
function restoreMatrix(rowSum: number[], colSum: number[]): number[][] {
    // we need to create two pointer and initializing the matrix 
    const n:number=rowSum.length;
    const m:number=colSum.length;

    const result:number[][]=Array.from({length:n},function(){ return Array(m).fill(0)});
    let i:number=0,j:number=0;
    while(i<n && j<m){
        let minele:number=Math.min(rowSum[i],colSum[j]);
        rowSum[i]-=minele;
        colSum[j]-=minele;
        result[i][j]=minele;
        if(rowSum[i]===0){
            i++;
        }else {
            j++;
        }
    }
return result;
};